lib/workers/repository/process/lookup/generate.ts

```diff
@@ -0,0 +1,68 @@
+import { Release } from '../../../../datasource';
+import { logger } from '../../../../logger';
+import type { LookupUpdate } from '../../../../manager/types';
+import { VersioningApi } from '../../../../versioning';
+import { LookupUpdateConfig } from './types';
+import { getUpdateType } from './update-type';
+
+export function generateUpdate(
+  config: LookupUpdateConfig,
+  versioning: VersioningApi,
+  currentVersion: string,
+  bucket: string,
+  release: Release
+): LookupUpdate {
+  const newVersion = release.version;
+  const update: LookupUpdate = {
+    bucket,
+    newVersion,
+    newValue: null,
+  };
+  const releaseFields = [
+    'checksumUrl',
+    'downloadUrl',
+    'newDigest',
+    'releaseTimestamp',
+  ];
+  for (const field of releaseFields) {
+    if (release[field] !== undefined) {
+      update[field] = release[field];
+    }
+  }
+  const { currentValue, rangeStrategy } = config;
+  try {
+    update.newValue = versioning.getNewValue({
+      currentValue,
+      rangeStrategy,
+      currentVersion,
+      newVersion,
+    });
+  } catch (err) /* istanbul ignore next */ {
+    logger.warn(
+      { err, currentValue, rangeStrategy, currentVersion, newVersion },
+      'getNewValue error'
+    );
+    update.newValue = currentValue;
```

Comment:
i think we should have a test for this

Reply:
I tried to trigger it but there's too much validation before it to prevent it. Should only occur if there's a mistake in a versioning implementation.

Reply:
So we can add tests for this function exclusively, so we can trigger this error. Lets do it in seperate pr 🙃